test/async-function.js

```diff
@@ -103,83 +93,77 @@ describe('Evaluate concurrent expressions with callbacks', function() {
             }
         };
 
-        expr.evaluate({"payload":1}, {}, function(err,result) {
+        expr.evaluate({"payload":1}, {}).then(function(result) {
             expect(result).to.equal('goat');
-            partDone();
-        });
-        expr.evaluate({"payload":2}, {}, function(err,result) {
+            return partDone();
+        // eslint-disable-next-line no-console
+        }).catch(e => console.error(e));
+
+
+        expr.evaluate({"payload":2}, {}).then(function(result) {
             expect(result).to.equal('cheese');
-            partDone();
-        });
+            return partDone();
+        // eslint-disable-next-line no-console
+        }).catch(e => console.error(e));
     });
 });
 
 describe('Handle chained functions that end in promises', function() {
-
-    var counter = function(count) {
-        return {
-            inc: function() {
-                return counter(count + 1);
-            },
-            then: function(onFulfilled) {
-                setImmediate(function(){
-                    onFulfilled(count);
-                });
-                return Promise.resolve(count);
-            }
-        };
-    };
+    const counter = async (count) => ({
```

Comment:
curious, why the `counter` function itself is `async`? it looks like only its `inc` property should be `async`?

Reply:
ah, nevermind - it's used within its own property